src/cpp/StatisticsBackend.cpp

```diff
@@ -182,11 +179,19 @@ EntityId create_and_register_monitor(
         const DomainParticipantQos& participant_qos,
         const DomainId domain_id = 0)
 {
+    // NOTE: This method is quite awful to read because of the error handle of every entity
+    // This could be done much nicer encapsulating this in Monitor creation in destruction, but you know...
+    // Why do not call stop_monitor in error case?, youll ask. Well, mutexes are treated rarely here, as this static
+    // class locks and unlocks StatisticsBackendData mutex, what makes very difficult to do some coherent
+    // calls from one to another.
+    // What should happen is that all this logic is moved to StatisticsBackendData. You know, some day...
```

Comment:
Why not doing it today?
This PR is already moving `stop_monitor`, so moving this method also seems logical.

Reply:
Time reasons. If we agree on applying these changes now, I'll do it.

Reply:
As before, I'd do it on a follow up